### DISCUSSION TITLE
GraphQL.Execution.QueryMissingError: GraphQL query is missing."

### DISCUSSION THREAD

**Discussion by Workaholic01** [graphql-dotnet/server#999] [Q&A]
I am trying to setup GraphQL playground but I am facing this error.

![image](https://user-images.githubusercontent.com/29353695/222136422-544da880-4b1b-4265-a679-c760914398e0.png)

When I inspect it lead me to this one

`"errors":[{"message":"GraphQL query is missing.","extensions":{"code":"QUERY_MISSING","codes":["QUERY_MISSING"],"details":"GraphQL.Execution.QueryMissingError: GraphQL query is missing."}}]}`

My Configurations look like as follows

`// Add GraphQL
```
	services.AddGraphQL(graphBuilder =>
		graphBuilder
			.AddSchema<GraphSchema>()
			.ConfigureExecutionOptions(options =>
			{
				var settings = options.RequestServices!.GetRequiredService<IOptions<GraphQLSettings>>().Value;
				var logger = options.RequestServices!.GetRequiredService<ILogger<Program>>();
				options.EnableMetrics = settings.EnableMetrics;
				options.UnhandledExceptionDelegate = (UnhandledExceptionContext context) => GraphExceptionHandler.HandleExceptionAsync(logger, context);
			})
			.AddSystemTextJson()
			.AddErrorInfoProvider((opts, serviceProvider) =>
			{
				IOptions<GraphQLSettings> settings = serviceProvider.GetRequiredService<IOptions<GraphQLSettings>>();
				opts.ExposeExceptionDetails = settings.Value.ExposeExceptions;
			})
			.AddGraphTypes(typeof(GraphSchema).Assembly)
			//.AddWebSocketAuthentication<WebSocketAuthenticationService>()
			.AddAuthorizationRule()
			.AddUserContextBuilder(GraphUserContextBuilder.Build)
	);`
```

```

app.UseGraphQL<GraphSchema>(graphSettings.Path,
		options =>
		{
			System.Console.WriteLine("options.HandleWebSockets: {0}", options.HandleWebSockets);
		});

	if (app.Environment.IsDevelopment() || app.Environment.IsStaging())
	{
		app.UseGraphQLPlayground(graphSettings.PlaygroundPath, new PlaygroundOptions
		{
			GraphQLEndPoint = graphSettings.Path,
			SubscriptionsEndPoint = graphSettings.WebSocketsPath,
			Headers = new Dictionary<string, object>
			{
				{
					HeaderNames.Authorization, "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJodHRwOi8vc2NoZW1hcy54bWxzb2FwLm9yZy93cy8yMDA1LzA1L2lkZW50aXR5L2NsYWltcy9uYW1lIjoic3poQHJlbGF0ZWl0LmNvbSIsImh0dHA"
				}
			}
		});
	}
```

I don't understand what am I missing here? Any help would be appreciated.

**Comment by Shane32** [comment]
Which GraphQL.NET nuget packages have you installed?

Can you copy and paste the payload that is sent to your server from Playground?  Preferably, headers also (no need for authorization header).

**Comment by Workaholic01** [comment]
Hi @Shane32 

Below are the GraphQL versions. And I am actually not sending any payload so far. Just trying to view the schema. Its not even loading that.

![image](https://user-images.githubusercontent.com/29353695/222345258-f615a3db-7299-480f-8d7f-ffc156b476bf.png)

**Comment by sungam3r** [comment]
I think some server exception may be in your logs.

**Comment by Shane32** [comment]
Playground is sending an introspection request via GraphQL in order to display the schema.  You can use the F12 key in Chrome to view the network logs, and then view the request, response, headers, etc.

**Comment by Shane32** [comment]
Just FYI, `GraphQL.Server.Authorization.AspNetCore` nuget package is only needed when calling the obsolete `AddGraphQLAuthorization()`.  The `AddAuthorizationRule` method is contained within `GraphQL.Server.Transports.AspNetCore`.  In other words, for typical use, `GraphQL.Server.Authorization.AspNetCore` is an obsolete package that is no longer needed.

**Comment by Workaholic01** [comment]
Yes, In the network tab I was able to figure out the issue. It was throwing an exception because I was using nested List Objects and I didn't register it as types. 
Thank you for all the help. :)
